src/Visitor/MutationsCollectorVisitor.php

```diff
@@ -58,6 +58,9 @@ public function leaveNode(Node $node)
     {
         foreach ($this->mutators as $mutator) {
             $isOnFunctionSignature = $node->getAttribute(WrappedFunctionInfoCollectorVisitor::IS_ON_FUNCTION_SIGNATURE, false);
+            if ($node->getAttribute(WrappedFunctionInfoCollectorVisitor::IS_ON_ABSTRACT_FUNCTION_SIGNATURE, false)) {
```

Comment:
I would say that checking `$node->isAbstract()` in `PublicVisibility.php` is enough to fix theoriginal issue.

I mean, checking this flag for each node is slows down the things without a benefit since it isn't even used.

Also, after merging https://github.com/infection/infection/pull/67 it would be better to store `[\ReflectionMethod](http://php.net/manual/ru/class.reflectionmethod.php)` that "knows" everything about particular method rather than just flag that it is an abstract one

Reply:
The goal of `IS_ON_ABSTRACT_FUNCTION_SIGNATURE` is to stop mutation of default values in abstract function signatures, since those will also never be covered.

For example:
```diff
- abstract public function find(bool $includeArgs = true): string;
+ abstract public function find(bool $includeArgs = false): string;
```

Will never be covered, and is therefore a useless mutation. 

But as i said before, if #67 gets merged this isn't really needed